includes/internal/tools/class-progress-tables-eraser.php

```diff
@@ -0,0 +1,187 @@
+	exit;
+}
+
+/**
+ * Class Progress_Tables_Eraser.
+ *
+ * @internal
+ *
+ * @since $$next-version$$
+ */
+class Progress_Tables_Eraser implements Sensei_Tool_Interface, Sensei_Tool_Interactive_Interface {
+	/**
+	 * Nonce action.
+	 *
+	 * @var string
+	 */
+	const NONCE_ACTION = 'sensei-tools-progress-tables-eraser';
+
+	/**
+	 * Initialize the tool.
+	 */
+	public function init(): void {
+		add_filter( 'sensei_tools', [ $this, 'register_tool' ] );
+	}
+
+	/**
+	 * Register the tool.
+	 *
+	 * @param array $tools List of tools.
+	 *
+	 * @return array
+	 */
+	public function register_tool( $tools ) {
+		$tools[] = $this;
+		return $tools;
+	}
+
+	/**
+	 * Get the ID of the tool.
+	 *
+	 * @return string
+	 */
+	public function get_id() {
+		return 'progress-tables-eraser';
+	}
+
+	/**
+	 * Get the name of the tool.
+	 *
+	 * @return string
+	 */
+	public function get_name() {
+		return __( 'Delete student progress tables', 'sensei-lms' );
+	}
+
+	/**
+	 * Get the description of the tool.
+	 *
+	 * @return string
+	 */
+	public function get_description() {
+		return __( 'Delete student progress and quiz submission tables. This will delete those tables, but won\'t affect comment-based data.', 'sensei-lms' );
+	}
+
+	/**
+	 * Run the tool.
+	 */
+	public function process() {
+		if ( empty( $_POST['delete-tables'] ) ) {
+			return;
+		}
+
+		$wpnonce = isset( $_POST['_wpnonce'] ) ? sanitize_text_field( wp_unslash( $_POST['_wpnonce'] ) ) : '';
+		if ( empty( $wpnonce ) || ! wp_verify_nonce( wp_unslash( $wpnonce ), self::NONCE_ACTION ) ) {
+			Sensei_Tools::instance()->trigger_invalid_request( $this );
+			return;
+		}
+
+		if ( empty( $_POST['confirm'] ) ) {
+			Sensei_Tools::instance()->add_user_message( __( 'You must confirm the action before it can be performed.', 'sensei-lms' ), true );
+			wp_safe_redirect( $this->get_tool_url() );
+			exit;
+		}
+
+		global $wpdb;
+
+		$results = array();
+		foreach ( $this->get_tables() as $table ) {
+			// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
+			if ( $wpdb->get_var( $wpdb->prepare( 'SHOW TABLES LIKE %s', $table ) ) === $table ) {
+				// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared, WordPress.DB.DirectDatabaseQuery.SchemaChange
+				$wpdb->query( "DROP TABLE $table" );
+				$results[] = $table;
+			}
+		}
+
+		if ( count( $results ) > 0 ) {
+			$message = sprintf(
+				/* translators: %s: list of tables. */
+				__( 'The following tables have been deleted: %s', 'sensei-lms' ),
+				implode( ', ', $results )
+			);
+
+			/**
+			 * Fires after progress tables are deleted.
+			 *
+			 * @since $$next-version$$
+			 *
+			 * @param array $tables List of deleted tables.
+			 */
+			do_action( 'sensei_tools_progress_tables_deleted', $results );
+		} else {
+			$message = __( 'No tables were deleted.', 'sensei-lms' );
+		}
+
+		Sensei_Tools::instance()->add_user_message( $message );
+		wp_safe_redirect( $this->get_tool_url() );
```

Comment:
I've noticed that the tool is unavailable after deleting the tables and you get a `This tool is not currently available. Please try again later.` notice and you are back to the tools main screen.

![image](https://github.com/Automattic/sensei/assets/1612178/08f5c9ba-913b-4fed-ba15-f8da8d6d9755)

To avoid this, maybe just redirect to the tools main screen instead?

Reply:
Thanks, Miro. Fixed it here: https://github.com/Automattic/sensei/pull/7230/commits/4299161b35697503f8c42ddff9fa53947b247cad

I redirect to the Tools page when tables were deleted.